modules/migrations/gitlab.go

```diff
@@ -0,0 +1,500 @@
+	gr, _, err := g.client.Projects.GetProject(g.repoID, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	var private bool
+	switch gr.Visibility {
+	case gitlab.InternalVisibility:
+		private = true
+	case gitlab.PrivateVisibility:
+		private = true
+	case gitlab.PublicVisibility:
+		private = false
+	default:
+		private = true
+	}
+	// convert gitlab repo to stand Repo
+	return &base.Repository{
+		//Owner:       gr.Owner.Username,
+		Name:        gr.Name,
+		IsPrivate:   private,
+		Description: gr.Description,
+		OriginalURL: gr.WebURL,
+		CloneURL:    gr.HTTPURLToRepo,
+	}, nil
+}
+
+// GetTopics return gitlab topics
+func (g *GitlabDownloader) GetTopics() ([]string, error) {
+	//r, _, err := g.client.Repositories.Get(g.ctx, g.repoOwner, g.repoName)
+	gr, _, err := g.client.Projects.GetProject(g.repoID, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	return gr.TagList, err
+}
+
+// GetMilestones returns milestones
+func (g *GitlabDownloader) GetMilestones() ([]*base.Milestone, error) {
+	var perPage = 100
+	var state = "all"
+	var milestones = make([]*base.Milestone, 0, perPage)
+	for i := 1; ; i++ {
+		ms, _, err := g.client.Milestones.ListMilestones(g.repoID, &gitlab.ListMilestonesOptions{
+			State: &state,
+			ListOptions: gitlab.ListOptions{
+				Page:    i,
+				PerPage: perPage,
+			}}, nil)
+		if err != nil {
+			return nil, err
+		}
+		var milestones = make([]*base.Milestone, 0)
+
+		for _, m := range ms {
+			var desc string
+			if m.Description != "" {
+				desc = m.Description
+			}
+			var state = "open"
+			if m.State != "" {
+				state = m.State
+			}
+			deadline, err := time.Parse("2006-01-02", m.DueDate.String())
+			if err != nil {
+				return nil, err
+			}
+			milestones = append(milestones, &base.Milestone{
+				Title:       m.Title,
+				Description: desc,
+				Deadline:    &deadline,
+				State:       state,
+				Created:     *m.CreatedAt,
+				Updated:     m.UpdatedAt,
+				Closed:      m.UpdatedAt,
+			})
+		}
+		if len(ms) < perPage {
+			break
+		}
+	}
+	return milestones, nil
+}
+
+// GetLabels returns labels
+func (g *GitlabDownloader) GetLabels() ([]*base.Label, error) {
+	var perPage = 100
+	var labels = make([]*base.Label, 0, perPage)
+	for i := 1; ; i++ {
+		ls, _, err := g.client.Labels.ListLabels(g.repoID, &gitlab.ListLabelsOptions{
+			Page:    i,
+			PerPage: perPage,
+		}, nil)
+		if err != nil {
+			return nil, err
+		}
+		for _, label := range ls {
+			baseLabel := &base.Label{
+				Name:        label.Name,
+				Color:       strings.TrimLeft(label.Color, "#)"),
+				Description: label.Description,
+			}
+			labels = append(labels, baseLabel)
+		}
+		if len(ls) < perPage {
+			break
+		}
+	}
+	return labels, nil
+}
+
+func (g *GitlabDownloader) convertGitlabRelease(rel *gitlab.Release) *base.Release {
+
+	r := &base.Release{
+		TagName:         rel.TagName,
+		TargetCommitish: rel.Commit.ID,
+		Name:            rel.Name,
+		Body:            rel.Description,
+		//Draft:           *rel.Draft,
```

Comment:
Same as above re: commented out code

Reply:
Thanks! Think I got all of them. 
Added a comment to the math being done to fetch the PR comments too.